engine/apps/mattermost/tests/test_tasks.py

```diff
@@ -0,0 +1,214 @@
+    mock_response = httpretty.Response(json.dumps(data), status=status.HTTP_200_OK)
+    httpretty.register_uri(httpretty.PUT, url, responses=[mock_response])
+
+    on_alert_group_action_triggered_async(ack_log_record.pk)
+
+    last_request = httpretty.last_request()
+    assert last_request.method == "PUT"
+    assert last_request.url == url
+
+    request_body = json.loads(last_request.body)
+    ids = [a["id"] for a in request_body["props"]["attachments"][0]["actions"]]
+    for id in ids:
+        assert id in expected_button_ids
+
+
+@pytest.mark.django_db
+@httpretty.activate(verbose=True, allow_net_connect=False)
+def test_on_alert_group_action_triggered_async_fails_without_alert_group_message(
+    make_organization_and_user,
+    make_alert_receive_channel,
+    make_alert_group,
+    make_alert,
+    make_mattermost_channel,
+    make_alert_group_log_record,
+):
+    organization, _ = make_organization_and_user()
+    alert_receive_channel = make_alert_receive_channel(organization)
+    ack_alert_group = make_alert_group(
+        alert_receive_channel, acknowledged_at=timezone.now() + timezone.timedelta(hours=1), acknowledged=True
+    )
+    make_alert(alert_group=ack_alert_group, raw_request_data=alert_receive_channel.config.example_payload)
+    make_mattermost_channel(organization=organization, is_default_channel=True)
+    ack_log_record = make_alert_group_log_record(ack_alert_group, type=AlertGroupLogRecord.TYPE_ACK, author=None)
+
+    with pytest.raises(MattermostMessage.DoesNotExist):
+        on_alert_group_action_triggered_async(ack_log_record.pk)
+
+
+@pytest.mark.django_db
+@httpretty.activate(verbose=True, allow_net_connect=False)
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"status_code": 400},
+        {"status_code": 401},
+    ],
+)
+def test_on_alert_group_action_triggered_async_failure(
+    make_organization_and_user,
+    make_alert_receive_channel,
+    make_alert_group,
+    make_alert,
+    make_mattermost_channel,
+    make_alert_group_log_record,
+    make_mattermost_message,
+    make_mattermost_post_response_failure,
+    data,
+):
+    organization, _ = make_organization_and_user()
+    alert_receive_channel = make_alert_receive_channel(organization)
+    ack_alert_group = make_alert_group(
+        alert_receive_channel, acknowledged_at=timezone.now() + timezone.timedelta(hours=1), acknowledged=True
+    )
+    make_alert(alert_group=ack_alert_group, raw_request_data=alert_receive_channel.config.example_payload)
+    make_mattermost_channel(organization=organization, is_default_channel=True)
+    ack_log_record = make_alert_group_log_record(ack_alert_group, type=AlertGroupLogRecord.TYPE_ACK, author=None)
+    mattermost_message = make_mattermost_message(ack_alert_group, MattermostMessage.ALERT_GROUP_MESSAGE)
+
+    url = "{}/api/v4/posts/{}".format(settings.MATTERMOST_HOST, mattermost_message.post_id)
+    data = make_mattermost_post_response_failure(status_code=data["status_code"])
```

Comment:
It looks like you only need to parametrize on the status code? (wondering why having a dict) also, you are overriding the data from the parametrize here, which can be confusing